hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/compress/TestCodecFactory.java

```diff
@@ -137,9 +138,15 @@ private static CompressionCodecFactory setClasses(Class[] classes) {
   
   private static void checkCodec(String msg, 
                                  Class expected, CompressionCodec actual) {
-    assertEquals(msg + " unexpected codec found",
-                 expected.getName(),
-                 actual.getClass().getName());
+    if (expected == null) {
```

Comment:
Is there any test case that passes null for expected?

Reply:
There are 4 total, testing for cases where the codec should not exist in class.